components/camera/align/extrinsics.go

```diff
@@ -146,10 +146,15 @@ func newColorDepthExtrinsics(ctx context.Context, color, depth camera.Camera, at
 		debug:     attrs.Debug,
 		logger:    logger,
 	}
+	var cameraModel transform.PinholeCameraModel
```

Comment:
this can probably be a refactored function if attrs is the same type across each of these camera instances

Reply:
I agree, turn this into a helper function in camera.go and you can use it in all of the camera models.

Reply:
Unfortunately they are not always the same. `attrs` or `params` are one of these types at these locations:
```
*extrinsicsAttrs: rdk/components/camera/align/extrinsics.go:118
*joinAttrs: rdk/components/camera/align/join.go:102
*homographyAttrs: rdk/components/camera/align/homography.go:104
camera.Properties: components/camera/camera.go:134 & in the `transformpipeline` package & in slam service
*transformConfig: rdk/components/camera/transformpipeline/pipeline.go:77
*dualServerAttrs: rdk/components/camera/videosource/server.go:111
*ServerAttrs: rdk/components/camera/videosource/server.go:246
*fileSourceAttrs: rdk/components/camera/videosource/static.go:28
*WebcamAttrs: rdk/components/camera/videosource/webcam.go
```

As a result I've made the helper function take 2 params, one for each `PinholdCameraModel` struct member. Let me know if that works or if I'm missing something / if you would prefer I add a function per type `attrs` / `params` type.